app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,793 @@
+          onDragLeave={this.endDragover}
+          onDrop={this.drop}>
+          <div className="dropzone-msg">
+            <p>150 MB filesize limit</p>
+            <p>Only .csv files will import correctly</p>
+            <p>Fine grained import options are only for files under 200KB</p>
+          </div>
+          {this.fileLimitLink("Close")}
+        </div>
+      );
+    },
+
+    render: function () {
+      var box = this.defaultBox();
+
+      if (this.state.draggingOver) {
+        box = this.boxIsDraggingOver();
+      } else if (this.state.loading) {
+        box = this.boxIsLoading();
+      } else if (this.state.showLimitInfo) {
+        box = this.boxShowLimitInfo();
+      }
+      return box;
+    }
+  });
+
+  var DataImporterPreviewData = React.createClass({
+    getInitialState: function () {
+      return {
+        left: 0
+      };
+    },
+    bigFilePreviewWarning: function () {
+      var rowShown = this.props.rowShown,
+          totalRows = this.props.rowsTotal;
+
+      return (
+        <div className="top-row">
+          <div className="big-file-info-message">
+            <p className="big-file-preview-limit-info-message">
+              Because of the size of this file, this preview only shows the
+              first {rowShown} rows, out of {totalRows} rows total.
+              However, if you choose to load the data into a database, the
+              entirety of the file (all {totalRows} rows) will be imported.
+            </p>
+          </div>
+          {this.loadingIntoDB()}
+        </div>
+      );
+    },
+    componentDidMount: function () {
+      document.getElementById('preview-page')
+        .addEventListener('scroll', this.handleScroll);
+    },
+    componentWillUnmount: function () {
+      document.getElementById('preview-page')
+        .removeEventListener('scroll', this.handleScroll);
+    },
+    handleScroll: function (e) {
+      this.setState({left: document.getElementById('preview-page').scrollLeft});
+    },
+
+    loadingIntoDB: function () {
+      if (this.props.isloadingInDBInProgress) {
+        return (
+          <div className="dataIsLoading">
+            <Components.LoadLines />
+          </div>
+        );
+      }
+      return null;
+    },
+
+    fileMetadataInfo: function () {
+      var totalRows = this.props.rowsTotal,
+          previewMsg = totalRows > 500 ? 'This is a 500 row (max) preview of the file.' : '';
+
+      return (
+        <div className="top-row">
+          <div className="big-file-info-message">
+            <p className="big-file-preview-limit-info-message">
+              All {totalRows} rows from this file will be imported. {previewMsg}
+            </p>
+          </div>
+          {this.loadingIntoDB()}
+        </div>
+      );
+    },
+    render: function () {
+
+      var fileInfoMessage =
+            this.props.isBigFile ? this.bigFilePreviewWarning() :  this.fileMetadataInfo(),
+          style = {'left': this.state.left};
```

Comment:
this is super tricky. So I am using jquery to move the top controls to the center of the page whenever we scroll left and right, so the 'the.state.left.' is the variable, it changes from 0 to n. A class won't work, but the CSS I had tried was not really good either. One option would be to have a floating element that happens sometimes, but then it breaks more than it helps.